code/modules/mob/living/simple_animal/hostile/megafauna/onestarboss.dm

```diff
@@ -66,70 +424,217 @@ Make jump sequience and call related proc "great_leap_forward"
-		B.attack_generic(src,rand(melee_damage_lower,melee_damage_upper),attacktext)
-		return B
-
-/mob/living/simple_animal/hostile/megafauna/one_star/proc/shoot_rocket(turf/marker, set_angle)
-	if(!isnum(set_angle) && (!marker || marker == loc))
-		return
-	var/turf/startloc = get_turf(src)
-	var/obj/item/projectile/P = new /obj/item/projectile/bullet/rocket(startloc)
-	P.firer = src
-	if(target)
-		P.original = target
-	P.launch( get_step(marker, pick(SOUTH, NORTH, WEST, EAST, SOUTHEAST, SOUTHWEST, NORTHEAST, NORTHWEST)) )
-
+/mob/living/simple_animal/hostile/megafauna/one_star/AttackingTarget() // overwriting the thing because mech loves to bite people like roaches do. Erased for the pattern-based combat.
+	OpenFire()
+//	if(!Adjacent(target_mob))
+	//	return
+	//if(isliving(target_mob))
+//		var/mob/living/L = target_mob
+	//	L.attack_generic(src,rand(melee_damage_lower,melee_damage_upper),attacktext)
+	//	return L
+	//if(istype(target_mob, /mob/living/exosuit))
+	//	var/mob/living/exosuit/M = target_mob
+	//	M.attack_generic(src,rand(melee_damage_lower,melee_damage_upper),attacktext)
+	//	return M
+//	if(istype(target_mob,/obj/machinery/bot))
+	//	var/obj/machinery/bot/B = target_mob
+//		B.attack_generic(src,rand(melee_damage_lower,melee_damage_upper),attacktext)
+//		return B
 
 /mob/living/simple_animal/hostile/megafauna/one_star/OpenFire()
-	anger_modifier = CLAMP(((maxHealth - health)/50),0,20)
-	ranged_cooldown = world.time + 120
-	walk(src, 0)
-	telegraph()
-	icon_state = "onestar_boss"
-	if(prob(35))
-		shoot_rocket(target_mob.loc, rand(0,90))
-		move_to_delay = initial(move_to_delay)
-		MoveToTarget()
-		return
-	if(prob(45))
-		select_spiral_attack()
-		move_to_delay = initial(move_to_delay)
-		MoveToTarget()
-		return
-	if(target_mob)
-		if(prob(75))
-			wave_shots()
-			move_to_delay = initial(move_to_delay)
-			MoveToTarget()
-			return
-		else
-			shoot_projectile(target_mob.loc, rand(0,90))
-			MoveToTarget()
-	move_to_delay = initial(move_to_delay)
\ No newline at end of file
+	if(!doing_something)
+		if(src.stat != DEAD && target_mob)
+			if(!move_lock)
+				action = pick(move_list)
+			switch(action)
+				if(OS_BOSS_SHOTGUN)
+					var/obj/effect/effect/telegraph/F = new /obj/effect/effect/telegraph(target_mob.loc)
+					move_lock = TRUE
+					for(F in orange(src, 30))
+						if(F in orange(1, src)) // this is fucking stupid, but if you remove this and stand on the hologram it will not be able to kick you for some reason, see Bump()
+							for(var/mob/living/targ in orange(1, src))
+								if(F.loc == targ.loc)
+									qdel(F)
+									//targ.gib() // I havent come for a better solution due to laziness and inability to fix.
+									targ.Weaken(3)
+									//playsound(targ.loc, 'sound/mechs/obliteration.ogg', 100) DEAL LOOK
+									var/kick_dir = get_dir(src, targ)
+									src.UnarmedAttack(targ, rand(50,100), attacktext)
+									targ.throw_at(get_edge_target_turf(targ, kick_dir), 3, 1)
```

Comment:
```suggestion
									var/kick_dir = get_dir(src, targ)
									UnarmedAttack(targ, rand(50,100))
									targ.throw_at(get_edge_target_turf(targ, kick_dir), 3, 1)
```
Saw these exact lines before in the PR, consider reusing code instead of copypasting.